applications/facemask/libs/detectors/jetson/openpifpaf_tensorrt/openpifpaf_tensorrt.py

```diff
@@ -0,0 +1,230 @@
+        result = []
+
+     
+        for i, pred_object in enumerate(predictions):
+            pred = pred_object.data
+            pred_visible = pred[pred[:, 2] > .2]
+            xs = pred_visible[:, 0]
+            ys = pred_visible[:, 1]
+            
+            if len(xs) == 0 or len(ys) == 0:
+                continue
+
+            x, y, w, h = pred_object.bbox()
+
+            x_min = int(x)
+            x_max = int(x + w)
+            y_min = int(y)
+            y_max = int(y + h)
+            xmin = int(max(x_min - .15 * w, 0))
+            xmax = int(min(x_max + .15 * w, self.w))
+            ymin = int(max(y_min - .2 * h, 0))
+            ymax = int(min(y_max + .05 * h, self.h))
+            bbox_dict={}
+            
+            # extract face bounding boxi
+            if np.all(pred[[0, 1, 2, 5, 6], -1] > 0.15):
+                x_min_face = int(pred[6, 0])
+                x_max_face = int(pred[5, 0])
+                y_max_face = int((pred[5, 1] + pred[6, 1]) / 2)
+                y_eyes = int((pred[1, 1] + pred[2, 1]) / 2)
+                y_min_face = 2 * y_eyes - y_max_face
+                if (y_max_face - y_min_face > 0) and (x_max_face - x_min_face > 0):
+                    h_crop = y_max_face - y_min_face
+                    x_min_face = int(max(0, x_min_face - 0.05 * h_crop))
+                    y_min_face = int(max(0, y_min_face - 0.05 * h_crop))
+                    x_max_face = int(min(self.w, x_min_face + 1.1 * h_crop))
+                    y_max_face = int(min(self.h, y_min_face + 1.1 * h_crop))
+                    bbox_dict["bbox"] = [y_min_face / self.h, x_min_face / self.w, y_max_face / self.h, x_max_face / self.w]
+            else: 
+                x_min_head = self.w
+                y_min_head = self.h
+                x_max_head = 0
+                y_max_head = 0
+                for i in range(6):
+                    if pred[i,0] > 0.0:
+                        x_min_head = min(x_min_head, pred[i,0])
+                    if pred[i,1] > 0.0:
+                        y_min_head = min(y_min_head, pred[i,1])
+                    x_max_head = max(x_max_head, pred[i,0])
+                    y_max_head = max(y_max_head, pred[i,1])
+                    h_crop = y_max_head - y_min_head
+                if ( x_min_head != self.w and x_max_head != 0 and y_min_head != self.h and y_max_head != 0 and x_min_head != x_max_head and y_min_head != y_max_head ):
+                    x_min_head = int(max(0, x_min_head - 0.2 * h_crop))
+                    y_min_head = int(max(0, y_min_head - 0.4 * h_crop))
+                    x_max_head = int(min(self.w, x_max_head + 1 * h_crop))
+                    y_max_head = int(min(self.h, y_max_head + 0.8 * h_crop))
+
+                    bbox_dict["bbox_head"] = [y_min_head / self.h, x_min_head / self.w, y_max_head / self.h, x_max_head / self.w]
+            result.append(bbox_dict)
+        return result
```

Comment:
Can you please apply this logic in other detectors?

Reply:
Sure.